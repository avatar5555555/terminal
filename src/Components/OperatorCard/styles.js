// @flow

import type { Theme } from 'src/types'

const styles = (theme: Theme) => ({
  image: {
    width: 100,
    height: 100,
    margin: theme.spacing.unit
  },
  container: {
    display: 'flex',
    flex: 1
  },
  root: {
    width: '100%'
  },
  icon: {
    marginLeft: theme.spacing.unit
  },
  button: {
    alignSelf: 'center',
    [theme.breakpoints.down('sm')]: {
      alignSelf: 'flex-end'
    }
  },
  content: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    flex: 1,
    [theme.breakpoints.down('sm')]: {
      flexDirection: 'column',
      alignItems: 'flex-end'
    }
  }
})

export default styles
