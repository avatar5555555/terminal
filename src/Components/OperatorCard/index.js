// @flow

import React from 'react'
import Grid from '@material-ui/core/Grid'
import CardMedia from '@material-ui/core/CardMedia'
import { withStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import Send from '@material-ui/icons/Send'
import { FormattedMessage } from 'react-intl'
import { Link } from 'react-router-dom'

type Props = {
  src: string,
  name: string,
  classes: {
    image: string,
    container: string,
    root: string,
    icon: string,
    button: string,
    content: string
  }
}

const styles = theme => ({
  image: {
    width: 100,
    height: 100,
    margin: theme.spacing.unit
  },
  container: {
    display: 'flex',
    flex: 1
  },
  root: {
    width: '100%'
  },
  icon: {
    marginLeft: theme.spacing.unit
  },
  button: {
    alignSelf: 'center',
    [theme.breakpoints.down('sm')]: {
      alignSelf: 'flex-end'
    }
  },
  content: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    flex: 1,
    [theme.breakpoints.down('sm')]: {
      flexDirection: 'column',
      alignItems: 'flex-end'
    }
  }
})

const OperatorCard = ({ src, name, classes }: Props) => (
  <Grid item xs={12} md={6} lg={4} className={classes.root}>
    <Card>
      <div className={classes.container}>
        <CardMedia image={src} title={name} className={classes.image} />
        <CardContent className={classes.content}>
          <Typography variant="button">
            <FormattedMessage id={`Operator.${name}`} />
          </Typography>
          <Button
            variant="contained"
            color="primary"
            className={classes.button}
            component={Link}
            to={{ pathname: '/payment', state: { name } }}
          >
            <FormattedMessage id="Actions.pay" />
            <Send className={classes.icon} />
          </Button>
        </CardContent>
      </div>
    </Card>
  </Grid>
)

export default withStyles(styles)(OperatorCard)
