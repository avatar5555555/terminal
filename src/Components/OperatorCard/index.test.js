import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import OperatorCard, { styles } from '.'

const props = {
  src: 'string',
  name: 'string',
  classes: {
    image: 'string',
    container: 'string',
    root: 'string',
    icon: 'string',
    button: 'string',
    content: 'string'
  }
}

describe('OperatorCard', () => {
  const wrap = (props = {}) => shallow(<OperatorCard {...props} />)

  it('renders', () => {
    expect(toJSON(wrap(props))).toMatchSnapshot()
  })

  it('renders without crashing', () => {
    expect(wrap(props)).toBeDefined()
  })

  it('styles return object', () => {
    const classes = styles({
      spacing: { unit: 5 },
      breakpoints: { down: jest.fn() }
    })

    expect(classes.image.margin).toBe(5)
  })

  it('checks if Grid is item', () => {
    const item = wrap(props)
      .dive()
      .find('.OperatorCard-root-3')
      .prop('item')

    expect(item).toBeTruthy()
  })
})
