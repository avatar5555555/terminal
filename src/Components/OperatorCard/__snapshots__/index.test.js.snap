// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatorCard renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(OperatorCard)
    classes={
        Object {
            "button": "string",
            "container": "string",
            "content": "string",
            "icon": "string",
            "image": "string",
            "root": "string",
          }
    }
    name="string"
    src="string"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "button": "OperatorCard-button-5 string",
        "container": "OperatorCard-container-2 string",
        "content": "OperatorCard-content-6 string",
        "icon": "OperatorCard-icon-4 string",
        "image": "OperatorCard-image-1 string",
        "root": "OperatorCard-root-3 string",
      },
      "name": "string",
      "src": "string",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "button": "OperatorCard-button-5 string",
          "container": "OperatorCard-container-2 string",
          "content": "OperatorCard-content-6 string",
          "icon": "OperatorCard-icon-4 string",
          "image": "OperatorCard-image-1 string",
          "root": "OperatorCard-root-3 string",
        },
        "name": "string",
        "src": "string",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
